src/shared/components/data-table/DataTable.test.tsx

```diff
@@ -0,0 +1,121 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import { times } from 'lodash';
+
+import DataTable from './DataTable';
+import { defaultDataTableState } from 'src/shared/components/data-table/dataTableReducer';
+
+import { type TableProps } from 'src/shared/components/data-table/DataTable';
+
+import { TableData, DataTableColumns } from './dataTableTypes';
+
+export const createDataTableSampleData = (
+  rows: number,
+  columns: number
+): { data: TableData; columns: DataTableColumns } => {
+  return {
+    data: times(rows, (row) =>
+      times(columns, (column) => `Cell ${row},${column}`)
+    ),
+    columns: times(columns, (column) => ({
+      columnId: `${column}`,
+      title: `Column ${column}`,
+      helpText: `Column ${column} help text`,
+      isSortable: true
```

Comment:
Which of these fields are optional? Since you have a test for rendering without errors, I would use it to render the table with a just a minimum required amount of props.